packages/monaco/src/split-editor.tsx

```diff
@@ -178,6 +178,50 @@ export default function SplitEditor({
     }
   }, [monaco]);
 
+  useEffect(() => {
+    // validates user code when user code is directly manipulated without going through editorState (e.g. reset button)
+    const monacoAndEditorStateReady = monaco && userEditorState;
+    async function typeCheck() {
+      if (monacoAndEditorStateReady) {
+        const models = monaco.editor.getModels();
+        const getWorker = await monaco.languages.typescript.getTypeScriptWorker();
+
+        for (const model of models) {
+          const worker = await getWorker(model.uri);
+          const diagnostics = (
+            await Promise.all([
+              worker.getSyntacticDiagnostics(model.uri.toString()),
+              worker.getSemanticDiagnostics(model.uri.toString()),
+            ])
+          ).reduce((a, b) => a.concat(b));
+
+          const markers = diagnostics.map((d) => {
+            const start = model.getPositionAt(d.start!);
+            const end = model.getPositionAt(d.start! + d.length!);
+
+            return {
+              severity: monaco.MarkerSeverity.Error,
+              startLineNumber: start.lineNumber,
+              endLineNumber: end.lineNumber,
+              startColumn: start.column,
+              endColumn: end.column,
+              message: d.messageText as string,
+            } satisfies monaco.editor.IMarkerData;
+          });
+
+          monaco.editor.setModelMarkers(model, model.getLanguageId(), markers);
+        }
+      }
+    }
+
+    return () => {
+      if (monacoAndEditorStateReady) {
+        typeCheck();
+        onMount?.tests?.(userEditorState, monaco);
+      }
+    };
+  }, [userCode, tests]);
```

Comment:
does monaco and userEditorState need to be added to this? not sure if anything else is missing

Reply:
re: monaco and userEditorState -- yeah onMount's tests are expecting 2 arguments 

![Screenshot 2023-09-14 at 1 54 25 AM](https://github.com/typehero/typehero/assets/28593720/cc52e8d3-5d78-453b-a7c2-5d0a0b4557ed)

re: triggering on other cases -- you mean if it validates the tests again on each input change? yah I think so


https://github.com/typehero/typehero/assets/28593720/e8748c50-d69a-439b-8294-bc223ea7ad26

Reply:
i just want to verify we're not running the typecheck code twice per keystroke or any other events and that the effect only triggers on the reset code case

Reply:
im asking if they needed to be added to the dep array because you are using them in your useEffect. im assuming you'd get a lint error so maybe its fine

Reply:
you're right, it is evaluated 2 times. let me work out some more fixes

Reply:
all good lmk when its ready!

Reply:
nevermind -- I was just dumb. this should be good now @bautistaaa